### PR TITLE
GroupController deletes (deactivates) groups and deletes future events

### PR DESCRIPTION
**Pull request by BSteephenson**
When you 'delete' a group
- active becomes false
- future events get deleted
- when you call GET /user it only populates groups that are active

^ This is super cool of you to do. Let's make this customary. +1 --Alex

### CODE REVIEW

**Comment by lyra833** [comment]
Watching this get coded has convinced me to switch to test driven development holy jesus.

**Comment by zfrenchee** [comment]
Hey @BSteephenson, I snuck in a commit which opened spaces after each of those function declarations, which you've removed in this PR. I personally like having them there. We should consider deciding on a style guide more formally, or just informally deciding here. 

I think my personal bias comes from the fact that code scares me, and I like having a lower density of the stuff. So generally I want code to occupy roughly 30% of the screen real estate. If it occupies less, that's obviously inefficient, and if it occupies more, I have a hard time teasing apart what it does, oftentimes. 

Whitespace is a sort of code annotation that uses negative space to influence the eyes of the reader and imply what is and what isn't quite as important, or what is related to what. I really like GroupController's airy-ness with those line breaks after the function declarations. If you strongly disagree I'm sure we can put together a consensus style-guide. If you don't feel too strongly about it, let's just add them back for now.

File: tuftslife/api/controllers/GroupController.js [hunk 1]
```diff
@@ -85,4 +84,22 @@ module.exports = {
 		});		//res.json({message: 'Success', });
```
Comment (zfrenchee) [inline tuftslife/api/controllers/GroupController.js]: I feel like we can remove this at this point? Is there any reason to keep this commented response?

File: tuftslife/api/controllers/GroupController.js [hunk 2]
```diff
@@ -85,4 +84,22 @@ module.exports = {
...
 	},
 
+	delete : function(req, res) {
+		if (!req.isAuthenticated()) return res.json({message: "User not logged in"})
+		Group.findOne({name: req.param('name')}).populate('admins').exec(function (err, group){
+			// TODO handle error
```
Comment (zfrenchee) [inline tuftslife/api/controllers/GroupController.js]: What don't you handle here? The case where the Group isn't returned from the db? That should definitely be filled in before we merge this PR.

File: tuftslife/api/controllers/GroupController.js [hunk 3]
```diff
@@ -85,4 +84,22 @@ module.exports = {
...
+			if(!group) return res.json({message: "Group does not exist"})
```
Comment (zfrenchee) [inline tuftslife/api/controllers/GroupController.js]: Check FindOne's behavior. If it doesn't find a group, does it return undefined, or does it raise an error. You might be duplicating work here?

File: tuftslife/api/controllers/GroupController.js [hunk 4]
```diff
@@ -85,4 +84,22 @@ module.exports = {
...
+			if(_.contains(_.pluck(group.admins, 'id'), req.user.id) === false){
+				return res.json({message: "User is not an admin of this group"}) }
+
+			Event.destroy({owner: group.id, endTime : {'>=': new Date()} } ).exec(function(err, events){
+				group.active = false;
+				group.save(function(err, group){
+					res.json(events)
+				})
```
Comment (zfrenchee) [inline tuftslife/api/controllers/GroupController.js]: This last chunk is semantically peculiar. Inside the call to destroy events, you only operate on the group. It makes some sense, on second thought, but I wonder why nothing is being done with the 'events' object in this function's scope. 

Additionally, you should check the err object first, ofc.

**Comment by zfrenchee** [comment]
A very elegant PR. Nice work.

**Comment by BSteephenson** [comment]
Most of my deleting whitespace was putting console.logs there and then deleting them afterwards, thinking I put the whitespace there so deleting it =P

File: tuftslife/api/controllers/GroupController.js (hunk 2 above)
Comment (BSteephenson) [inline tuftslife/api/controllers/GroupController.js]: I do in the next like with the error we don't get a group. I'll get rid of the //TODO

File: tuftslife/api/controllers/GroupController.js (hunk 4 above)
Comment (BSteephenson) [inline tuftslife/api/controllers/GroupController.js]: That was an arbitrary decision. I just needed to make sure that deactivating the group and deleting the past events happened before sending a response. Stupid callbacks. I'll reverse it so it looks like it makes more sense

File: tuftslife/api/controllers/GroupController.js (hunk 3 above)
Comment (BSteephenson) [inline tuftslife/api/controllers/GroupController.js]: "If no matching record is found, the value of found will be undefined. Not finding a record does not constitute an error for findOne."

**Comment by BSteephenson** [comment]
Updated
- adds whitespace in both controllers
- adds an error handler with Group.findOne
- reorder the function calls so you deactive the group and then delete the events (prettier code)

File: tuftslife/api/controllers/GroupController.js [hunk 5]
```diff
@@ -82,7 +82,29 @@ module.exports = {
...
+			if(!group) return res.json({message: "Group does not exist"})
+
+			if(_.contains(_.pluck(group.admins, 'id'), req.user.id) === false){
+				return res.json({message: "User is not an admin of this group"}) }
+
+			group.active = false;
+			group.save(function(err, group){
+				//delete all events of this group after the current time
+				Event.destroy({owner: group.id, endTime : {'>=': new Date()} } ).exec(function(err, events){
+					res.json(events)
```
Comment (zfrenchee) [inline tuftslife/api/controllers/GroupController.js]: Let's just check the error object first, just in case.

**Comment by zfrenchee** [comment]
Last thing, then we'll merge.

**Comment by BSteephenson** [comment]
✓

**Pull request merged**